### DISCUSSION TITLE
Suggest Vitest as an option for Angular apps/libraries

### DISCUSSION THREAD

**Discussion by yjaaidi** [nrwl/nx#26339] [Feature Requests]
**TL;DR:** `@nx/angular:application` and `@nx/angular:library` generators should suggest Vitest as an option for testing.

Just as Nx played an important role in popularizing Jest in the Angular community, it is probably time for Nx to do the same for Vitest.

Luckily, @brandonroberts and the Analog team did a great job to successfully bridge the gap between Angular and Vite ecosystem with [`@analogjs/vite-plugin-angular`](https://github.com/analogjs/analog/tree/beta/packages/vite-plugin-angular) and [`@analogjs/vitest-angular`](https://github.com/analogjs/analog/tree/beta/packages/vitest-angular).

I'll be happy to submit a PR.

**Comment by juristr** [comment]
I like the idea 👍

**Comment by pavankjadda** [comment]
+1

**Comment by ErickRodrCodes** [comment]
If bridged properly with Vite thanks to Analog, Imagine what people can do : like, electron applications built on Angular through Vite. +1 with this

**Reply by brandonroberts** [reply]
I think @joshuamorony is already doing this 👀

**Comment by Yberion** [comment]
+1 for sure!

**Comment by luishcastroc** [comment]
This is a great idea I’m in

**Comment by gergobergo** [comment]
I really like the idea, especially since Karma is deprecated, and other than Jest you don't have any proper (not deprecated) option for testing.

**Comment by polroyal** [comment]
I like the idea

**Comment by brandonroberts** [comment]
I'm excited to see this move forward. Until then, for anyone who wants to try out the `@analogjs/vitest-angular` package, use the Nx generator.

```sh
npm install @analogjs/platform --save-dev
```

```sh
ng g @analogjs/platform:setup-vitest --project [your-project-name]
```

Full guide - https://analogjs.org/docs/features/testing/vitest#using-a-schematicgenerator

**Comment by the-ult** [comment]
Created a follow-up feature-request for the `@nx/vite` plugin:
- https://github.com/nrwl/nx/discussions/28952
